experiment/measurer/test_experiment_coverage_utils.py

```diff
@@ -13,32 +13,301 @@
 # limitations under the License.
 """Tests for experiment_coverage_utils.py"""
 import os
+import pandas as pd
 
-from experiment.measurer import experiment_coverage_utils as exp_cov_utils
+from experiment.measurer import experiment_coverage_utils
 
 TEST_DATA_PATH = os.path.join(os.path.dirname(__file__), 'test_data')
 
+# Arbitrary values to use in tests.
+SUMMARY_JSON_FILE = 'cov_summary.json'
+NUM_FUNCTION_IN_COV_SUMMARY = 3
+NUM_COVERED_SEGMENTS_IN_COV_SUMMARY = 16
+FILE_NAME = "/home/test/fuzz_no_fuzzer.cc"
+FUNCTION_NAMES = ['main', '_Z3fooIfEvT_', '_Z3fooIiEvT_']
+
+# Arbitrary values for simple tests
+BENCHMARK = 'freetype2'
+FUZZER = 'afl'
+TRIAL_ID = 2
+TIMESTAMP = 900
+
+# Arbitrary values for testing experiment specific dataframe.
+BENCHMARKS = ['benchmark_1', 'benchmark_2']
+FUZZERS = ['fuzzer_1', 'fuzzer_2']
+TIMESTAMPS = [900, 1800]
+
 
 def get_test_data_path(*subpaths):
     """Returns the path of |subpaths| relative to TEST_DATA_PATH."""
     return os.path.join(TEST_DATA_PATH, *subpaths)
 
 
-def test_extract_segments_and_functions_from_summary_json(fs):
+def test_extract_segments_and_functions_from_summary_json_for_segments(fs):
+    """Tests that extract_covered_regions_from_summary_json returns the covered
+    segments and functions from summary json file."""
+
+    summary_json_file = get_test_data_path(SUMMARY_JSON_FILE)
+    fs.add_real_file(summary_json_file, read_only=False)
+
+    df_container = (experiment_coverage_utils.
+                    extract_segments_and_functions_from_summary_json(
+                        summary_json_file, BENCHMARK, FUZZER, TRIAL_ID,
+                        TIMESTAMP))
+
+    fuzzer_ids = df_container.segment_df['fuzzer'].unique()
+    benchmark_ids = df_container.segment_df['benchmark'].unique()
+    file_ids = df_container.segment_df['file'].unique()
+
+    assert len(df_container.segment_df) == NUM_COVERED_SEGMENTS_IN_COV_SUMMARY
+
+    for fuzzer_id in fuzzer_ids:
+        # Check if type recorded for the ID id is "fuzzer".
+        assert (df_container.name_df[df_container.name_df['id'] == fuzzer_id]
+                ['type'].item() == "fuzzer")
+        # Check if fuzzer ids match in "segment" data frame and "name" data
+        # frame.
+        assert (fuzzer_id == df_container.name_df[df_container.name_df['name']
+                                                  == FUZZER]['id'].item())
```

Comment:
As this repeats across 2 test cases x 3 types of ids, you can write a simple function for this code.

Reply:
Made a function for the same.